### PR TITLE
[GH-25492] Add the request context to all public methods in server/channels/app/admin.go

### PR DESCRIPTION
**Pull request by thomasbrq**
#### Summary
Add the request context to all public methods in `server/channels/app/admin.go`

- [x] Change all public methods in `server/channels/app/admin.go` to receive `rctx request.CTX` as the first argument [1].
- [x] Change all methods in `server/channels/app/admin.go` (even private ones) that use the mlog logger to use receive the context as well. The methods to look for are `mlog.Debug`,`mlog.Info`, `mlog.Warn`, `mlog.Error`.
- [x] Make sure all functions calling the changed methods pass along the context.
- [x] Update the generated interfaces with `make generated`.
- [x] Update the tests.

[1] Only if they have an `*App`receiver: `func (a *App) MethodName()`, others like `func (s *Server)` ... should be skipped.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/25492

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/admin.go [hunk 1]
```diff
@@ -111,23 +111,23 @@ func AddLocalLogs(logData map[string][]string, s *Server, page, perPage int, ser
...
+func (a *App) GetLogs(rctx request.CTX, page, perPage int) ([]string, *model.AppError) {
+	return a.Srv().GetLogs(rctx, page, perPage)
 }
 
 func (s *Server) GetLogsSkipSend(page, perPage int, logFilter *model.LogFilter) ([]string, *model.AppError) {
 	return s.platform.GetLogsSkipSend(page, perPage, logFilter)
 }
 
-func (a *App) GetLogsSkipSend(page, perPage int, logFilter *model.LogFilter) ([]string, *model.AppError) {
+func (a *App) GetLogsSkipSend(rctx request.CTX, page, perPage int, logFilter *model.LogFilter) ([]string, *model.AppError) {
```
Comment (hanzei) [inline server/channels/app/admin.go:126]: Let's blank these to make it clear that they are unused
```suggestion
func (a *App) GetLogsSkipSend(_ request.CTX, page, perPage int, logFilter *model.LogFilter) ([]string, *model.AppError) {
```

File: server/channels/app/admin.go [hunk 2]
```diff
@@ -111,23 +111,23 @@ func AddLocalLogs(logData map[string][]string, s *Server, page, perPage int, ser
...
 	return a.Srv().GetLogsSkipSend(page, perPage, logFilter)
 }
 
-func (a *App) GetClusterStatus() []*model.ClusterInfo {
+func (a *App) GetClusterStatus(rctx request.CTX) []*model.ClusterInfo {
```
Comment (hanzei) [inline server/channels/app/admin.go:130]:
```suggestion
func (a *App) GetClusterStatus(_ request.CTX) []*model.ClusterInfo {
```

File: server/channels/jobs/batch_migration_worker.go [hunk 3]
```diff
@@ -111,7 +111,8 @@ func (worker *BatchMigrationWorker) IsEnabled(_ *model.Config) bool {
 // checkIsClusterInSync returns true if all nodes in the cluster are running the same version,
 // logging a warning on the first mismatch found.
 func (worker *BatchMigrationWorker) checkIsClusterInSync() bool {
```
Comment (hanzei) [inline server/channels/jobs/batch_migration_worker.go:113]: Should `checkIsClusterInSync` accept a `request.CTX` instead?

**Review by hanzei — CHANGES_REQUESTED** [review]
Nice work :+1:

File: server/channels/jobs/batch_migration_worker.go (hunk 3 above)
Comment (thomasbrq) [reply server/channels/jobs/batch_migration_worker.go]: it looks better !

File: server/channels/app/admin.go (hunk 1 above)
Comment (hanzei) [reply server/channels/app/admin.go]: Can you please apply the same change to all method where `request.CTX` is unused?

**Comment by thomasbrq** [comment]
Should I do `make app-layers` ?

**Comment by hanzei** [comment]
Yes, you need to run that command, @thomasbrq

**Comment by thomasbrq** [comment]
> Yes, you need to run that command, @thomasbrq

Yes but we cannot use `_` as a parameter name in Go, we will have build errors @hanzei .

**Comment by hanzei** [comment]
Good catch @thomasbrq :+1: I filed https://mattermost.atlassian.net/browse/MM-55732 to fix the build issues. Would you please use `rctx` in the mean time? We can change it back to blanks later.

**Comment by thomasbrq** [comment]
> Good catch @thomasbrq 👍 I filed https://mattermost.atlassian.net/browse/MM-55732 to fix the build issues. Would you please use `rctx` in the mean time? We can change it back to blanks later.

done @hanzei

File: server/channels/jobs/batch_migration_worker.go [hunk 4]
```diff
@@ -110,8 +110,8 @@ func (worker *BatchMigrationWorker) IsEnabled(_ *model.Config) bool {
 
 // checkIsClusterInSync returns true if all nodes in the cluster are running the same version,
 // logging a warning on the first mismatch found.
-func (worker *BatchMigrationWorker) checkIsClusterInSync() bool {
-	clusterStatus := worker.app.GetClusterStatus()
+func (worker *BatchMigrationWorker) checkIsClusterInSync(rctx request.CTX) bool {
+	clusterStatus := worker.app.GetClusterStatus(rctx)
 	for i := 1; i < len(clusterStatus); i++ {
 		if clusterStatus[i].SchemaVersion != clusterStatus[0].SchemaVersion {
 			worker.logger.Warn(
```
Comment (hanzei) [inline server/channels/jobs/batch_migration_worker.go:117]: While we are here: 
```suggestion
			rctx.
			rctx.Logger().Warn(
```

**Review by hanzei — APPROVED** [review]
Nice work :rocket: 

Thanks @thomasbrq

**Review by streamer45 — APPROVED** [review]
Looks good :+1:

**Pull request merged**